adapt kubeclient changes

moolitayer opened the pull request:
adapt kubeclient changes from kubeclient pr#92[1]

adapt kubeclient changes from kubeclient pr#114[2]

[1] abonas/kubeclient#92
[2] abonas/kubeclient#114

Commit:
adapt kubeclient changes from kubeclient pr#92[1]

[1] https://github.com/abonas/kubeclient/pull/92

moolitayer commented:
marking as wip until kubeclient version is known

moolitayer commented:
fyi @abonas @simon3z

simon3z commented on lib/openshift_client.rb:
```diff
@@ -20,17 +20,50 @@ class Client < Kubeclient::Common::Client
 
+    private
+
+    def validate_auth_options(opts)
```
@moolitayer any way to reuse the kubeclient `validate_auth_options`? (instead of copy/paste)

moolitayer replied:
yes and we now need:
https://github.com/abonas/kubeclient/pull/109

moolitayer commented:
this now depends on abonas/kubeclient#109 too

moolitayer commented:
@aboans @simon3z  please review.

abonas commented on openshift_client.gemspec:
```diff
@@ -24,7 +24,7 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency 'webmock'
   spec.add_development_dependency 'rubocop', '= 0.30.0'
-  spec.add_dependency 'kubeclient', '>= 0.1.17'
+  spec.add_dependency 'kubeclient', '>= 0.3.1'
```
relevant gem ver is now 0.4.0 (released)

moolitayer replied:
updated

On Sun, Aug 23, 2015 at 2:32 PM, abonas notifications@github.com wrote:

> In openshift_client.gemspec
> https://github.com/abonas/openshift_client/pull/10#discussion_r37704999:
> 
> > @@ -24,7 +24,7 @@ Gem::Specification.new do |spec|
> >    spec.add_development_dependency 'minitest'
> >    spec.add_development_dependency 'webmock'
> >    spec.add_development_dependency 'rubocop', '= 0.30.0'
> > -  spec.add_dependency 'kubeclient', '>= 0.1.17'
> > -  spec.add_dependency 'kubeclient', '>= 0.3.1'
> 
> relevant gem ver is now 0.4.0 (released)
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/abonas/openshift_client/pull/10/files#r37704999.

moolitayer commented:
removing wip since kubeclient 0.4.0 has been released

moolitayer commented:
changed

```
-                   auth_options: {
-                     user: nil,
-                     password: nil,
-                     bearer_token: nil
-                   })
+                  auth_options: {})
```

to comply with verify_ssl

moolitayer commented:
fyi @abonas

abonas commented on lib/openshift_client.rb:
```diff
@@ -20,11 +20,28 @@ class Client < Kubeclient::Common::Client
-      ssl_options
+      @ssl_options = ssl_options
+
+      @basic_auth_user = auth_options[:user]
```
the code in lines 42-44 differs from similar code in kubeclient and provides a slightly different handling logic (and doesn't handle bearer token file), is this on purpose?
cc @oschreib @jimmidyson

moolitayer replied:
Right. 
This patch includes the mandatory changes openshift_client needed to not break 
using kubeclient after abonas/kubeclient#92
It does not included support for bearer token file added by @jimmidyson in https://github.com/abonas/kubeclient/pull/98

abonas replied:
> Right.
> This patch includes the mandatory changes openshift_client needed to not break
> using kubeclient after abonas/kubeclient#92
> It does not included support for bearer token file added by @jimmidyson in abonas/kubeclient#98

but even without bearer token file, the basic auth handling code is different - here it is always assigned.

moolitayer replied:
Pretty sure those changes where made in the same pr.

let me know if you need anything from me to move this patch.

abonas replied:
> Pretty sure those changes where made in the same pr.
> 
> let me know if you need anything from me to move this patch.

please look here
https://github.com/abonas/kubeclient/blob/master/lib/kubeclient.rb#L51
as you can see, there's a conditional there that checks if basic auth details should be assigned or not.
in your current patch here, you are always assigning basic auth details, while the bearer token is assigned conditionally - hence the logic is different.

moolitayer replied:
On Tue, Aug 25, 2015 at 2:50 PM, abonas notifications@github.com wrote:

> In lib/openshift_client.rb
> https://github.com/abonas/openshift_client/pull/10#discussion_r37856078:
> 
> > ```
> >    handle_uri(uri, path)
> >    @api_version = version
> >    @headers = {}
> > ```
> > -      ssl_options
> > -      @ssl_options = ssl_options
> >   +
> > -      @basic_auth_user = auth_options[:user]
> 
> Pretty sure those changes where made in the same pr.
> 
> let me know if you need anything from me to move this patch.
> 
> please look here
> https://github.com/abonas/kubeclient/blob/master/lib/kubeclient.rb#L51
> as you can see, there's a conditional there that checks if basic auth
> details should be assigned or not.
> in your current patch here, you are always assigning basic auth details,
> while the bearer token is assigned conditionally - hence the logic is
> different.

That change in logic was done as part of dyson bearer patch
(not saying it's related to the bearer patch but it was 'snuck in' together
)

so it is an additional change.

—

> Reply to this email directly or view it on GitHub
> https://github.com/abonas/openshift_client/pull/10/files#r37856078.

abonas commented:
@simon3z please also review. 
I've expressed my concern regarding the addition of different logic for basic auth handling 
vs the one present in kubeclient above.

simon3z commented:
@abonas @moolitayer let's move:

https://github.com/abonas/kubeclient/blob/master/lib/kubeclient.rb#L51-L59

to a common place (`common.rb` in kubeclient) and let's reuse it from here.

abonas commented:
> @abonas @moolitayer let's move:
> 
> https://github.com/abonas/kubeclient/blob/master/lib/kubeclient.rb#L51-L59
> 
> to a common place (common.rb in kubeclient) and let's reuse it from here.

:+1:

moolitayer commented:
marking as wip until https://github.com/abonas/kubeclient/pull/114 matures

Commit:
adapt kubeclient changes from kubeclient pr#114[1]

[1] https://github.com/abonas/kubeclient/pull/114

Commit:
change in dependency specification on kubeclient.

Use ruby's passimistic operator (~>). to allow only backward
compatible changes in kubeclient. Version is bumped to 5.0.x[1].

[1] https://rubygems.org/gems/kubeclient/versions/0.5.0

Commit:
increase max line length from 80 to 100

moolitayer commented:
> marking as wip until abonas/kubeclient#114 matures

removing wip, kubeclient 0.5.0 with abonas/kubeclient#114  has been released.

simon3z commented:
The auth logic is now completely shared with kubeclient. Merging as master is currently broken without this PR.
